src/DevChatter.Bot.Web/wwwroot/js/sprite.js

```diff
@@ -1,24 +1,119 @@
+  this.mod = mod || this.getRandomModifier(); //sets rate of displacement (how fast the sprite is moving)
   this.timesRendered = 0;
+  this.bounced = false;
+  
 };
 
 Sprite.prototype.getRandomDirection = function () {
-  var rand = Math.random() * 2 * Math.PI //get random val between 0 and 2 Pi ~ 0 and 360 degrees
+  var rand = Math.random() * 2 * Math.PI; //get random val between 0 and 2 Pi ~ 0 and 360 degrees
   
   return rand;
 };
 
-Sprite.prototype.update = function () {
-  var mod = 2
+Sprite.prototype.getRandomModifier = function () {
+  var rand = Math.random() * 3 + 1; //get random value between 1 and 4 (float)
+  return rand;
+};
+
+Sprite.prototype.update = function (canvas) {
   //assuming right is 0 and left is pi rad, though doesn't really matter
   //mod ~ modifier affects magnitude of displacement
-  this.x += mod * Math.cos(this.direction) //apply degree to trig function for x displacement
-  this.y += mod * Math.sin(this.direction) //apply degree to trig function for y displacement
+
+  x_right_border = canvas.width - this.image.width; //width of canvas - width of image, insures the right border of the image bounces off the right border of the canvas
+  y_bottom_border = canvas.height - this.image.height; //same as above but for bottom of image and canvas
+  x_left_border = 0;
+  y_top_border = 0;
+
+  this.x += this.mod * Math.cos(this.direction); //apply degree to trig function for x displacement
+  this.y += this.mod * Math.sin(this.direction); //apply degree to trig function for y displacement
+
+  // screen coords are defined as 0,0 in the top left and max width, height in the bottom right
+
+
+  //check if x position has moved beyond the left or right border
+  if ((this.x < x_left_border || this.x > x_right_border) && !this.bounced) {
+    // this.direction += Math.PI / 2;
+    this.direction = this.getBounceAngle(this.direction, true);
+    this.bounced = true;
+  }
+
+  //check if y position has moved beyond the upper or lower border
+  if ((this.y < y_top_border || this.y > y_bottom_border) && !this.bounced) {
+    // this.direction += Math.PI / 2;
+    this.direction = this.getBounceAngle(this.direction, false);
+    this.bounced = true;
+  }
+
+  //not sure if this could be removed and replaced with a canvas.intersects(this.x, this.y) or something simmilar. Could also be moved to a intersects() function
+  if ((this.y < y_bottom_border && this.y > y_top_border) && (this.x > x_left_border && this.x < x_right_border)) {
+    this.bounced = false; //if sprite is within the box reset bounce flag
+  }
+
+};
+
+// takes angle of colision and a bool to tell which kind of wall it's bouncing off of
+// outputs the incident angle that the sprite will be going in
+Sprite.prototype.getBounceAngle = function (angle, vertical_wall = true) { 
+  var QUART_CIRCLE = Math.PI / 2;
+  var HALF_CIRCLE = Math.PI;
+  var FULL_CIRCLE = 2 * Math.PI;
+  var bounce_angle = 0;
+
+
+  //half baked math incoming
```

Comment:
Maybe I'm oversimplifying, but couldn't we get away with something as simple as this?
```
if (vertical_wall) {
   bounce_angle = Math.PI - angle;
} else {
   bounce_angle = Math.PI * 2 - angle;
}
```

Reply:
Yeah I think that would work, I knew there had to be a simpler way.